packages/sync-service/lib/electric/shapes/querying.ex

```
@@ -57,13 +64,18 @@ defmodule Electric.Shapes.Querying do
     {query, []}
   end
 
-  defp get_column_names(table_info, root_table) do
+  defp get_column_names(table_info, root_table, nil) do
     table_info
     |> Map.fetch!(root_table)
     |> Map.fetch!(:columns)
     |> Enum.map(& &1.name)
   end
 
+  defp get_column_names(table_info, root_table, selected_columns) do
```

Why do you need to filter the `selected_columns` out of the column names you get from table info?
The column names are just the `selected_columns` argument itself.
I understand that this function will only return the column names in `selected_columns` that really exist in the table, but should that case be allowed?
I think we should throw a validation error in the shape plug if the user selects some columns that don't actually exist.
Then, here we can rely on the `selected_columns` and no need to get them out of the table info.